codegen/src/main/java/com/paypal/selion/plugins/CodeGeneratorMojo.java

```diff
@@ -124,6 +125,11 @@ private String pathToFolder(File file) {
      */
     public void execute() throws MojoExecutionException {
 
+        // iff basePackage is null, empty, or just whitespace reset it to the default value
+        basePackage = StringUtils.defaultString(basePackage, "com.paypal.selion.testcomponents");
```

Comment:
@krmahadevan put a java file in `src/main/java` in a maven project. don't give it a `package` declaration. See what "package" the class ends up in -- `(default package)`.  Certainly `(default package)` != `package default;`  However, I still feel some sort of convention/solution is better than an exception here. Maven users are used to it applying conventions. 

We could also just change the velocity template to omit the `package` line when `basePackage` and `folder` are empty.